src/platform/Zephyr/GenericBLEManagerImpl_Zephyr.cpp

```diff
@@ -495,6 +501,13 @@ CHIP_ERROR GenericBLEManagerImpl_Zephyr<ImplClass>::HandleTXComplete(const ChipD
     return CHIP_NO_ERROR;
 }
 
+template <class ImplClass>
+void GenericBLEManagerImpl_Zephyr<ImplClass>::HandleBLEAdvertisementTimeout(System::Layer * layer, void * param, System::Error error)
+{
+    ConnectivityMgr().SetBLEAdvertisingEnabled(false);
```

Comment:
Perhaps this should just call BLEMgr().SetAdvertisingEnabled() not to introduce cyclic dependency between BLEManager and ConnectivityManager?

Reply:
Done